src/basic/cgroup-util.c

```diff
@@ -572,10 +547,23 @@ int cg_get_path_and_check(const char *controller, const char *path, const char *
         assert(controller);
         assert(fs);
 
-        /* Check if the specified controller is actually accessible */
-        r = controller_is_accessible(controller);
+        if (!cg_controller_is_valid(controller))
+                return -EINVAL;
+
+        r = cg_all_unified();
         if (r < 0)
                 return r;
+        if (r > 0) {
+                /* In the unified hierarchy all controllers are considered accessible,
+                 * except for the named hierarchies */
+                if (startswith(controller, "name="))
+                        return -EOPNOTSUPP;
```

Comment:
this dropped the logic of allowing `name=systemd` in the v2 case. On purpose?

Reply:
The reasoning is that in the unified mode, we work only with the v2 hierarchy and that doesn't allow/need any `name=`. So `name=systemd` should be bathwater too, not a baby.
The logic should be same as before the change, the relevant check was against 
> #define SYSTEMD_CGROUP_CONTROLLER "_systemd"

that itself doesn't start with `name=`.

@poettering Does that clarify?